x/metadata/keeper/query_server.go

```diff
@@ -409,25 +411,25 @@ func (k Keeper) Records(c context.Context, req *types.RecordsRequest) (*types.Re
 			// This assumes that we have checked and set scopeAddr while processing the recordAddr.
 			scopeAddr3 := recordAddr.MustGetAsScopeAddress()
 			if !scopeAddr2.Equals(scopeAddr3) {
-				return &retval, status.Errorf(codes.InvalidArgument, "session %s and record %s are not associated with the same scope", sessionAddr, recordAddr)
+				return &retval, sdkerrors.ErrInvalidRequest.Wrapf("session %s and record %s are not associated with the same scope", sessionAddr, recordAddr)
 			}
 		case scopeAddr.Empty():
 			scopeAddr = scopeAddr2
 		case !scopeAddr.Equals(scopeAddr2):
-			return &retval, status.Errorf(codes.InvalidArgument, "session %s is not part of scope %s", recordAddr, scopeAddr)
+			return &retval, sdkerrors.ErrInvalidRequest.Wrapf("session %s is not part of scope %s", recordAddr, scopeAddr)
 		}
 	}
 	if len(req.Name) > 0 {
 		if scopeAddr.Empty() {
 			// assumes scopeAddr is set previously while parsing other input.
-			return &retval, status.Error(codes.InvalidArgument, "a scope or session is required to look up records by name")
+			return &retval, sdkerrors.ErrInvalidRequest.Wrap("a scope or session is required to look up records by name")
 		}
 		// We know that scopeAddr is legit, and that we have a name. So this won't give an error.
 		recordAddr2 := scopeAddr.MustGetAsRecordAddress(req.Name)
 		if recordAddr.Empty() {
 			recordAddr = recordAddr2
 		} else if !recordAddr.Equals(recordAddr2) {
-			return &retval, status.Errorf(codes.InvalidArgument, "record %s does not have name %s", recordAddr, req.Name)
+			return &retval, sdkerrors.ErrNotFound.Wrapf("record %s does not have name %s", recordAddr, req.Name)
```

Comment:
Should this be: `sdkerrors.ErrInvalidRequest`. I see you are using Invalid request in other areas in the query server for checks like this that fail?  Or should those others be changed? Example: ```		recordAddr2 := scopeAddr.MustGetAsRecordAddress(req.RecordName)
		if recordAddr.Empty() {
			recordAddr = recordAddr2
		} else if !recordAddr.Equals(recordAddr2) {
			return &retval, sdkerrors.ErrNotFound.Wrapf("record %s does not have name %s", recordAddr, req.RecordName)
		} ```